File: examples/nodejs/mail/mail.js
var nodemailer = require('nodemailer');

exports.send = function(event, callback) {
  var url = process.env.MAIL_SMTP_URL
  var from = process.env.MAIL_FROM

  if (!event.data.to) return callback('email recipient ("to") not found')
  if (!event.data.subject) return callback('subject not found')
  if (!event.data.text) return callback('text not found')

  var mail = {
    from: from,
    to: event.data.to,
    subject: event.data.subject,
    text: event.data.text
  }

  nodemailer
    .createTransport(url)
    .sendMail(mail, function(error, resp) {
      if (error) return callback(error)
      callback(null, JSON.stringify(resp))
  });

}

File: examples/python/mail/mail.py
import os
import smtplib

from email.mime.text import MIMEText

def send(event):
  url = os.environ['MAIL_SMTP_URL']
  user = os.environ['MAIL_SMTP_USER']
  password = os.environ['MAIL_SMTP_PASSWORD']

  try:
    msg = MIMEText(event['data']['text'])
  except:
    raise Exception('no message found')

  msg['Subject'] = event['data']['subject']
  msg['From'] = user
  msg['To'] = event['data']['to']

  s = smtplib.SMTP(url)
  s.login(user, password)
  s.sendmail(user, event['data']['to'], msg.as_string())
  s.quit()

  return "mail sent to " + event['data']['to']